timeseries/src/autogluon/timeseries/models/presets.py

```diff
@@ -131,150 +177,137 @@ def get_preset_models(
-        logger.info(f"Excluded model types: {excluded_model_types}")
-        for model in excluded_model_types:
-            if not isinstance(model, str):
-                raise ValueError(f"Each entry in `excluded_model_types` must be a string, received {type(model)}")
-            excluded_models.add(normalize_model_type_name(model))
-    return excluded_models
-
-
-def get_hyperparameter_dict(
-    hyperparameters: Union[str, Dict[str, Union[ModelHyperparameters, List[ModelHyperparameters]]], None],
-    hyperparameter_tune: bool,
-) -> Dict[str, List[ModelHyperparameters]]:
-    hyperparameter_dict = {}
-
-    if hyperparameters is None:
-        hyperparameter_dict = copy.deepcopy(get_default_hps("default"))
-    elif isinstance(hyperparameters, str):
-        hyperparameter_dict = copy.deepcopy(get_default_hps(hyperparameters))
-    elif isinstance(hyperparameters, dict):
-        hyperparameter_dict = copy.deepcopy(hyperparameters)
-    else:
-        raise ValueError(
-            f"hyperparameters must be a dict, a string or None (received {type(hyperparameters)}). "
-            f"Please see the documentation for TimeSeriesPredictor.fit"
+        return model_type
+
+    def _get_default_model_init_kwargs(self) -> Dict[str, Any]:
+        return dict(
+            path=self.path,
+            freq=self.freq,
+            prediction_length=self.prediction_length,
+            eval_metric=self.eval_metric,
+            target=self.target,
+            quantile_levels=self.quantile_levels,
+            covariate_metadata=self.covariate_metadata,
         )
 
-    hyperparameter_dict = check_and_clean_hyperparameters(
-        hyperparameter_dict, must_contain_searchspace=hyperparameter_tune
-    )
-
-    return hyperparameter_dict
-
+    def _get_model_name(self, ag_args: Dict[str, Any], model_type: Type[AbstractTimeSeriesModel]) -> str:
+        name = ag_args.get("name")
+        if name is None:
+            name_stem = re.sub(r"Model$", "", model_type.__name__)
+            name_prefix = ag_args.get("name_prefix", "")
+            name_suffix = ag_args.get("name_suffix", "")
+            name = name_prefix + name_stem + name_suffix
+        return name
 
-def normalize_model_type_name(model_name: str) -> str:
-    """Remove 'Model' suffix from the end of the string, if it's present."""
-    if model_name.endswith("Model"):
-        model_name = model_name[: -len("Model")]
-    return model_name
 
-
-def check_and_clean_hyperparameters(
-    hyperparameters: Dict[str, Union[ModelHyperparameters, List[ModelHyperparameters]]],
-    must_contain_searchspace: bool,
-) -> Dict[str, List[ModelHyperparameters]]:
-    """Convert the hyperparameters dictionary to a unified format:
-    - Remove 'Model' suffix from model names, if present
-    - Make sure that each value in the hyperparameters dict is a list with model configurations
-    - Checks if hyperparameters contain searchspaces
+class HyperparameterBuilder:
```

Comment:
I'm a bit on the edge about moving this logic to two new classes. As far as I understand, both classes are essentially stateless (except the `banned_model_names`) and serve only for bundling the main public method with the helper methods. Could we achieve the same by 1/ adding a leading underscore to the methods that shouldn't be used outside of `models/presets.py` and 2/ exposing two public methods `resolve_hyperparameters` and `create_model_templates`?

Currently when I see 
```python
from .presets import HyperparameterBuilder, TrainableModelSetBuilder
```
I expect these classes to keep track of some state or each provide multiple methods, but then it turns out that each class only offers one public method that is called once.

Reply:
This looks great, thanks!